presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -314,6 +322,7 @@ protected Boolean visitInPredicate(InPredicate node, Void context)
         @Override
         protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
+            warnIfDistinctCount(node);
```

Comment:
`COUNT` is an aggregation function so you can add this check in the following block. And let's use standard libraries to check for function `FunctionResolution.isCountFunction`.